src/views/TradingCompetition/components/ShareImageModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useRef, useState, useEffect } from 'react'
+import { Modal, Flex, Button, Text } from '@pancakeswap-libs/uikit'
+import styled from 'styled-components'
+import useI18n from 'hooks/useI18n'
+import FlippersShare from '../pngs/flippers-share.png'
+import StormShare from '../pngs/storm-share.png'
+import CakersShare from '../pngs/cakers-share.png'
+import { localiseTradingVolume } from '../helpers'
+import { YourScoreProps } from '../types'
+
+const StyledCanvas = styled.canvas`
+  width: 100%;
+`
+
+const ShareImageModal: React.FC<YourScoreProps> = ({ onDismiss, profile, userLeaderboardInformation }) => {
+  const TranslateString = useI18n()
+  const { global, team, volume } = userLeaderboardInformation
+  const [bgImage, setBgImage] = useState(null)
+  const [profileImage, setProfileImage] = useState(null)
+  const canvas = useRef(null)
+
+  useEffect(() => {
+    const bgImages = [StormShare, FlippersShare, CakersShare]
```

Comment:
Main thing I'm looking for advice on here are two things I want to investigate tomorrow morning - 

1. Should I investigate some use of loadable to only import the image for the team that the user belongs to?

2. Currently - all of the images I'm pulling in and the canvas el I'm drawing on are full-size (1600 x 900), even though the preview within the modal is small. This is because when the image is downloaded, it converts this canvas to a png and downloads that. Ideally - I'd separate these so that the user only downloads the full-size canvas when they click the button. I'm gonna play around with this but if you have any thoughts, please share em!

Reply:
> Should I investigate some use of loadable to only import the image for the team that the user belongs to?

Yes. You can use the `useProfile()` hook to get the profile (which contains the user's team) from state.

>Currently - all of the images I'm pulling in and the canvas el I'm drawing on are full-size (1600 x 900), even though the preview within the modal is small. This is because when the image is downloaded, it converts this canvas to a png and downloads that. Ideally - I'd separate these so that the user only downloads the full-size canvas when they click the button. I'm gonna play around with this but if you have any thoughts, please share em!

I have never done anything like that but I think that is a good idea. Test it out if it is feasible but don't worry about it too much unless we continue to use this pattern.

Reply:
On 1. - my worry was that because I was importing each of the teams' raw share images, it would actually download each of these big bastards regardless of whether the users need them. But - it doesn't look like it does, so I think we're good. 

Testing with syrup storm user, it only downloads the `storm-share` asset

<img width="664" alt="Screenshot 2021-04-09 at 09 26 32" src="https://user-images.githubusercontent.com/79279477/114152185-e1683d00-9915-11eb-9501-66a68d063b11.png">

2. Good point on the limited utility unless we're going to reuse this component. I'm going to focus on getting this working on mobile ahead of this optimisation 👌